pkg/storage/wal/segment.go

```diff
@@ -144,6 +147,20 @@ func (b *SegmentWriter) Append(tenantID, labelsString string, lbls labels.Labels
 	}
 }
 
+func (b *SegmentWriter) Observe() {
+	b.consistencyMtx.Lock()
+	defer b.consistencyMtx.Unlock()
+
+	b.metrics.streams.Observe(float64(len(b.streams)))
+	tenants := make(map[string]struct{}, len(b.streams))
+	for _, s := range b.streams {
+		tenants[s.tenantID] = struct{}{}
+	}
+	b.metrics.tenants.Observe(float64(len(tenants)))
+	b.metrics.inputSizeBytes.Observe(float64(b.inputSize.Load()))
+	b.metrics.outputSizeBytes.Observe(float64(b.outputSize.Load()))
```

Comment:
Maybe we should only observe the output bytes if we've set a value here? In case we call Observe only after writing to the segment but haven't compressed/read it yet.

Reply:
Yeah! I was also thinking of documenting that `Observe` should be called after it has written the output, but idk?